akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ConsumerFilterRegistry.scala

```diff
@@ -35,27 +46,80 @@ import akka.projection.grpc.consumer.ConsumerFilter
   import ConsumerFilter._
   import ConsumerFilterRegistry._
 
-  private def behavior(subscribers: Set[Subscriber]): Behavior[Command] = {
-    Behaviors.receiveMessage {
-      case sub: Subscribe =>
-        // FIXME watch termination of subscriber, cleanup and such
+  private def behavior(
+      subscribers: Set[Subscriber],
+      stores: Map[String, ActorRef[ConsumerFilterStore.Command]]): Behavior[Command] = {
 
-        // FIXME should it be for a specific producer service? In the end that is only identified by the grpc client
-        //       settings. It would be possible several GrpcReadJournal for different producers but still with the same
-        //       streamId. Is it important to be able to filter those separately?
```

Comment:
In my thinking I have come to the conclusion that we should only use the streamId. The problem I was thinking about was that producer A and producer B (different regions) would use the same streamId and then consumer in C would setup streams to both A and B. The commands would then be propagated to both A and B.

My thinking is that if that is not desired one should use different streamId in A and B even though it may be the same service and same entityType.

Wdyt, @johanandren ?

Reply:
Hmmm. 

A replicated entity can anyway only be replicated to a consumer/replica "once" per cluster as we have the replica type-key name + self replica id + remote replica id as projection name.

But what about projections, this means that if you have multiple projections from the same service, filtering one of them will apply to all? And also if you both do replication and projection from the same service, that will always cover the same entity ids. Maybe that's fine, not sure.

Reply:
Maybe inconvenient, but then the producer side would have to expose them with several different streamIds even though they have the same underlying entityType. I'll create an issue for this, we must at least try it out and document it.